test/ConnectionTest4.cpp

```diff
@@ -0,0 +1,56 @@
+#include <gtest/gtest.h>
+
+#include "DBConnection4.hpp"
+
+namespace {
+
+    void checkMySqlConnection4Info(IDBConnection4& db_connection, const int expected_value) {
+        if (db_connection.connectionType() == ConnectionType::MYSQL) {
```

Comment:
Try to keep structure of is statements simpler. Also try to follow next idea, make asserts and checks first and continue if all fine
I suggest to rewrite this way
'''
If not mysqls then throw;
Assert cast
'''
Update here and bow

Reply:
@malirod , done